### PR TITLE
[API] placing order on multiple channels with api

### PR DESCRIPTION
**Pull request by arti0090**
This is a continuation of this PR #12091 
After the comments there were too many changes to do or revert so task will be continued here

### CODE REVIEW

File: src/Sylius/Behat/Context/Setup/ChannelContext.php [hunk 1]
```diff
@@ -30,12 +31,15 @@ final class ChannelContext implements Context
     /** @var SharedStorageInterface */
     private $sharedStorage;
 
+
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Setup/ChannelContext.php:34]:
```suggestion
```

File: src/Sylius/Behat/Context/Setup/ChannelContext.php [hunk 2]
```diff
@@ -30,12 +31,15 @@ final class ChannelContext implements Context
...
+    /** @var ChannelContextSetterInterface */
+    private $channelContextSetter;
+
     /** @var DefaultChannelFactoryInterface */
     private $unitedStatesChannelFactory;
 
     /** @var DefaultChannelFactoryInterface */
     private $defaultChannelFactory;
-
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Setup/ChannelContext.php:38]: Should be reverted

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Command/ChannelCodeAwareInterface.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php
+
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Command/ChannelCodeAwareInterface.php:2]: missing license block

**Review by AdamKasp — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, @arti0090! :1st_place_medal:

File: src/Sylius/Behat/Context/Setup/ChannelContext.php [hunk 4]
```diff
@@ -30,6 +31,9 @@ final class ChannelContext implements Context
     /** @var SharedStorageInterface */
     private $sharedStorage;
 
+    /** @var ChannelContextSetterInterface */
+    private $channelContextSetter;
```
Comment (Tomanhez) [inline src/Sylius/Behat/Context/Setup/ChannelContext.php:35]: Its needed? I guess it change channel in cookie. Api is not dependent on this.

**Pull request merged**